packages/expo-updates/android/src/main/java/expo/modules/updates/launcher/SelectionPolicyNewest.java

```diff
@@ -0,0 +1,67 @@
+package expo.modules.updates.launcher;
+
+import expo.modules.updates.db.entity.UpdateEntity;
+import expo.modules.updates.db.enums.UpdateStatus;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Simple Update selection policy which chooses
+ * the newest update (based on commit time) out
+ * of all the possible stored updates.
+ *
+ * If multiple updates have the same (most
+ * recent) commit time, this class will return
+ * the earliest one in the list.
+ */
+public class SelectionPolicyNewest implements SelectionPolicy {
+
+  private String mRuntimeVersion;
+
+  public SelectionPolicyNewest(String runtimeVersion) {
+    mRuntimeVersion = runtimeVersion;
+  }
+
+  @Override
+  public UpdateEntity selectUpdateToLaunch(List<UpdateEntity> updates) {
+    UpdateEntity updateToLaunch = null;
+    for (UpdateEntity update : updates) {
+      if (!mRuntimeVersion.equals(update.runtimeVersion)) {
+        continue;
+      }
+      if (updateToLaunch == null || updateToLaunch.commitTime.before(update.commitTime)) {
+        updateToLaunch = update;
+      }
+    }
+    return updateToLaunch;
+  }
+
+  @Override
+  public List<UpdateEntity> markUpdatesForDeletion(List<UpdateEntity> updates, UpdateEntity launchedUpdate) {
```

Comment:
Would it make sense to return an empty list early if `launchedUpdate` is null, and then we don't need to check if it's null later on?